File: iec61131lang/src/main/java/edu/kit/iti/formal/automation/datatypes/AnyReference.java
/*
 * #%L
 * iec61131lang
 * %%
 * Copyright (C) 2017 Alexander Weigl
 * %%
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public
 * License along with this program.  If not, see
 * <http://www.gnu.org/licenses/gpl-3.0.html>.
 * #L%
 */

package edu.kit.iti.formal.automation.datatypes;

import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * @author Augusto Modanese
 */
@Data
@EqualsAndHashCode
public class AnyReference extends AnyDt {
    public static final AnyReference ANY_REF = new AnyReference();

    @Override
    public String repr(Object obj) {
        throw new IllegalStateException("No repr for AnyReference");
    }

    @Override
    public <T> T accept(DataTypeVisitor<T> visitor) {
        return visitor.visit(this);
    }
}

File: iec61131lang/src/main/kotlin/edu/kit/iti/formal/automation/datatypes/AnyReference.kt
/*
 * #%L
 * iec61131lang
 * %%
 * Copyright (C) 2017 Alexander Weigl
 * %%
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a clone of the GNU General Public
 * License along with this program.  If not, see
 * <http://www.gnu.org/licenses/gpl-3.0.html>.
 * #L%
 */

package edu.kit.iti.formal.automation.datatypes

import lombok.Data
import lombok.EqualsAndHashCode

/**
 * @author Augusto Modanese
 */
@Data
@EqualsAndHashCode
open class AnyReference : AnyDt {

    override fun repr(obj: Any): String {
        throw IllegalStateException("No repr for AnyReference")
    }

    override fun <T> accept(visitor: DataTypeVisitor<T>): T? {
        return visitor.visit(this)
    }

    companion object {
        val ANY_REF = AnyReference()
    }
}
